docs: update graphql data-access retention docs (#24784)
MystenLabs/sui: docs/content/concepts/data-access/graphql-rpc.mdx

```graphql
query DynamicField {

> 	object(address: "0xb57fba584a700a5bcb40991e1b2e6bf68b0f3896d767a0da92e69de73de226ac") {

Author: Ah yeah, I thought ide was using a default linter but on second pass I did not see any configs for it in the docs. These are probably my global my prettier configs, i'll revert.

```graphql
{
	serviceConfig {
		maxQueryDepth
		maxQueryNodes
		maxOutputNodes
		defaultPageSize
		maxPageSize
		queryTimeoutMs
		maxQueryPayloadSize
		maxTypeArgumentDepth
		maxTypeArgumentWidth
		maxTypeNodes
		maxMoveValueDepth
	}
}
```

## Retention

Different queries are subject to their own retention policies. For example, live object set queries work only for recent checkpoints (measured in minutes or hours), while transaction pagination may be available only back to certain checkpoints (measured in weeks or months), depending on the filters applied. Data outside these ranges is pruned.

> ### Data Source Retention

Reviewer: ```suggestion
### Data source retention
```

GraphQL queries rely on different data sources with different retention ranges:

| Data Source          | Examples                                                  | Typical Retention          |
| -------------------- | --------------------------------------------------------- | -------------------------- |

> | **Consistent Store** | `Query.objects`, `Address.balance`, objects by owner/type | ~1 hour                    |

Reviewer: ```suggestion
| **Consistent store** | `Query.objects`, `Address.balance`, objects by owner or type | ~1 hour                    |
```

| **DB Store**         | `Query.transactions`, `Query.events`, `Query.checkpoints` | ~(30-90) days              |

> | **Archival Service** | Point lookups (tx by digest, object by ID+version)        | Indefinite (if configured) |

Reviewer: ```suggestion
| **Archival service** | Point lookups (transaction by digest, object by ID and version)        | Indefinite (if configured) |
```

:::tip Operators

Retention is configured by the stack operator. For production configuration examples, see [Indexer Stack Setup](/guides/operator/indexer-stack-setup).

:::

### Querying Available Range

Query the available checkpoint range using `serviceConfig.availableRange`:

```graphql
availableRange(type: String!, field: String, filters: [String!]): AvailableRange!
```

> - **`type`**: The GraphQL type name (e.g., `"Query"`, `"Address"`)

Reviewer: ```suggestion
- **`type`**: The GraphQL type name, such as `"Query"` or `"Address"`
```

			first {
				sequenceNumber
			}
			last {
				sequenceNumber
			}
		}
	}
}
```

```graphql
# Check available range for events filtered by emitting module
{
	serviceConfig {
		availableRange(type: "Query", field: "events", filters: ["module"]) {
			first {
				sequenceNumber
			}
			last {
				sequenceNumber
			}
		}
	}
}
```

:::info Nested queries

Each field in a nested query has its own retention limit:

```graphql
query {
	address(address: "0x...") {
		# Uses Consistent Store (~1 hour retention)
		dynamicFields(first: 10) {
			nodes {
				value {
					... on MoveObject {
						# Uses DB Store (may have different retention)
						objectAt(checkpoint: 10579000) {
							version
						}
					}
				}
			}
		}
	}
}
```

Use `serviceConfig.availableRange` to check retention for each field separately.

:::

### Handling "Outside Available Range" Errors

If you encounter an error indicating data is outside the available range, it means your query is requesting data that has been pruned. To resolve this:

1. **Check the available range first** using `serviceConfig.availableRange` before starting long pagination runs.

> 2. **Use fresh cursors** - old cursors may reference checkpoints that have since been pruned.

Reviewer: ```suggestion
2. **Use fresh cursors**. Old cursors may reference checkpoints that have since been pruned.
```
